app/controllers/user_profiles_controller.rb

```diff
@@ -3,29 +3,24 @@
 # Controller for create and edit UserProfile
 class UserProfilesController < ApplicationController
   before_action :set_info, only: %i[new create edit]
-  # before_action ... , only: %i[index new create edit update] # logged in?
+  before_action :signed_in, only: %i[index new create show edit update]
   # before_action ... , only: %i[edit update] # correct user?
 
   def index
     @user_profiles = UserProfile.all.page(params[:page])
   end
 
   def new
-    @user = User.new
-    # @user_profile = current_user.build_user_profile
-    @user_profile = @user.build_user_profile
+    if current_user.profile_registered?
+      redirect_to user_profiles_path # temp
+    else
```

Comment:
リダイレクトされるなら、
```suggestion
    end
```

でもいけたりしますかね？後続の処理のネストが減るので、できるならelse削りたいですね。

Reply:
redirect_toは最後の方に書けば方がいいみたいので、if文の条件を逆にしました。
https://guides.rubyonrails.org/layouts_and_rendering.html#using-redirect-to

Reply:
Rubcopが通らない…元に戻します。

Reply:
redirect_to XXX and returnという方法があるけど、使ったことがないです。
念のため、こうします。
https://stackoverflow.com/questions/34842427/redirect-to-vs-redirect-to-and-return